test/functional/aggregation_tests.js

```diff
@@ -2,6 +2,89 @@
 var expect = require('chai').expect;
 
 describe('Aggregation', function() {
+  it('Should use secondary readPreference if given', {
+    metadata: { requires: { topology: 'replicaset' } },
+
+    // The actual test we wish to run
+    test: function(done) {
+      const configuration = this.configuration;
```

Comment:
let's try to avoid using `configuration` as much as possible, we have a ticket to remove it outright. Instead, you should just require the `ReadPreference` normally above from the top level of the `mongodb` package